### ISSUE TITLE
Update to 1.89

### ISSUE THREAD

**Issue by calvertdw** [SpaiR/imgui-java#145]
Looks like ocornut fixed some docking and viewport issues upstream, which I know will help me out a lot.
https://github.com/ocornut/imgui/releases/tag/v1.89

**Comment by ds58** [comment]
I started to work on this today. Will need to update some of the submodules to be compatible.

**Comment by SpaiR** [comment]
@calvertdw I do work on #132 and there was a plan to publish a Dear ImGui update with it. Yet, right now, I can't find a spare time to complete the thing... If you need something specific, I recommend to fork the project and make proper changes in it. CI should work fine to get required jar's/dll's/so files.

**Comment by calvertdw** [comment]
@SpaiR I haven't looked too much into what you're doing but I figured I'd throw it out there: Do you thing JavaCPP would address some of your needs? We've used it a lot very successfully. https://github.com/bytedeco/javacpp

**Comment by SpaiR** [comment]
@calvertdw I've looked over solutions like JavaCPP and 3rd party java libraries is not what I want to pull into the binding. And, basically, they are unneeded, since the JNI layer is generated. The problem here is to manually write all the code needed for that. So, instead I'm writing a general purpose code generator which will be integrated with https://github.com/cimgui/cimgui.

tldr; I'm writing a code generator for a code generator to generate JNI code. 😅
